### PR TITLE
Staging Sites: Remove Upgrades and Inbox from navigation

### PR DESCRIPTION
**Pull request by kozer**
Fixes 1881-gh-Automattic/dotcom-forge

## Proposed changes:

'Upgrades' and 'Inbox' aren't relevant on Staging Sites:

image

This pr makes sure that these items are hidden for staging sites.

**NOTE:**  This pr needs D103951-code and 1303-gh-Automattic/wpcomsh to be merged first

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

Internal: pdtkmj-1iU-p2

## Does this pull request change what data or activity we track or use?

* No

#### **Pre-test:**

1. Create a WoA dev blog.
3. Enable ssh and sync this pr to the site.  
    ```
     jetpack rsync jetpack <ssh URL>:~/htdocs/wp-content/plugins/jetpack
     ```
4. Rsync changes from this pr 1303-gh-Automattic/wpcomsh to your WoA site (Insructions can be found in PCYsg-mkw-p2 )
5. Follow the instructions in D103951-code to set the persist-data

### Menu items appear correctly for non-staging sites

1. Visit your site:
     - `https://wordpress.com/hosting-config/<site>`
     - `<site>/wp-admin`
2. Make sure that the menu items appear.
3. Use the instructions in D103951-code to update the `wpcom_is_staging_site` (After apply the diff, change 1 to 0)
4. Reload both pages. Ensure that the menu items still appears

### Menu items don't appear for staging sites

1. Visit your site:
     - `https://wordpress.com/hosting-config/<site>`
     - `<site>/wp-admin`
2. Use the instructions in D103951-code to update the `wpcom_is_staging_site`
3. Reload both pages. Ensure that the menu items are not appearing

## Some more info about this PR.

To create a staging site we need the following:

- Have a WoA site.
- Create a staging site out of it.

So, all the staging sites have as a starting point a site that is WoA.

Also hosting config, which is the menu that enables a user to create a site, appears under the two following use cases:

- `WPcom_Admin_Menu`: This menu renders on **simple** sites.
- `Atomic_Admin_Menu`: This menu renders on **atomic** sites.

For simple sites, the staging creation is not available (The user has to enable hosting config first, which transfers the site to atomic)

Given all that, this pr takes the following things into consideration:

While we load the sidebar, we have 6 different use cases:

1. The site is a "Do It For Me" site (difm).
2. The site is a Jetpack site.
3. The site is a WoA site.
4. The site is a simple site.
7. The site is a P2 site.
8. The site is a domain-only site.

### **The site is a "Do It For Me" site (difm).**

Difm sites don't have those menu (Upgrades, inbox) items by default. So we **don't need to handle** that case. They also don't have a hosting configuration setting menu, so a user can't create a staging site for them. 

### **The site is a Jetpack site.**

Jetpack sites don't have a hosting configuration setting menu. A user can't create a staging site out of them. So we **don't need to handle** that case.

### **The site is a WoA site**
We **handle** that case, as when we create a staging site, the site goes Atomic.

### **The site is a simple site.**
A simple site can't be a staging site. (More info can be found in D103681-code). We **don't handle** that case.

### **The site is a P2 site.**
A P2 site can't be a staging site. We **don't handle** that case.

### **The site is a domain-only site.**
A domain-only site can't be a staging site. We **don't handle** that case.

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php [hunk 1]
```diff
@@ -69,7 +69,10 @@ public function reregister_menu_items() {
 		parent::reregister_menu_items();
 
 		$this->add_my_home_menu();
-		$this->add_inbox_menu();
+
+		if ( ! get_option( 'is_staging_site' ) ) {
```
Comment (danielbachhuber) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: This option will be blown away if the user uploads a database.

Let's use Atomic Persistent Data instead, and reference a `WPCOM_IS_STAGING_SITE` bit.

**Comment by danielbachhuber** [comment]
Worth noting: we actually hid these options with https://github.com/Automattic/wpcomsh/pull/1303.

We'll want to eventually land this PR (as it's the more robust, long-term approach). However, testing it will require spinning up and applying a new wpcomsh PR that removes the wpcomsh workaround.

**Comment by kozer** [comment]
From discussion with @danielbachhuber  about when to merge this pr:

It would be nice to land in the next week or two. However, we'll need to:
**1. Spin up a new wpcomsh pull request that reverts the wpcomsh code, and test against that.
2. Land this pull request.
3. Wait for Jetpack to be deployed to Atomic.
4. Land the wpcomsh pull request.**

**Comment by jeherve** [comment]
I struggled a bit with the instructions (couldn't figure out what the staging site that was created from my WoA dev site was, maybe none was created because my WoA dev site is still using a wpcomstaging.com domain?), so couldn't really test this. The changes do make sense though. 

I'll let others take a look, maybe someone will have better luck than me :)

**Review by wojtekn — APPROVED** [review]
@jeherve @kozer I've created PR that removes wpcomsh code 1312-gh-Automattic/wpcomsh and tested that change:

1. Created WoA dev blog
2. Applied the staging site sticker and ran code from D103951-code to persist data to the WoA site
3. Applied wpcomsh code from branch to WoA dev site using rsync
4. Applied this Jetpack branch to WoA site using Jetpack Beta
5. Confirmed that menus are gone from Calypso and WP Admin for this site

**Review by sejas — COMMENTED** [review]
@kozer, Thanks for this change! 🙌

**Pull request merged**